GiEnJul/Controllers/AdminController.cs

```diff
@@ -132,6 +132,58 @@ public async Task<ActionResult> PutWishAsync(string person_rowkey, [FromBody] st
             return Ok();
         }
 
+        [HttpDelete("Connection")]
+        [Authorize(Policy = "DeleteConnection")]
+        public async Task<ActionResult> DeleteConnectionAsync(string location, string rowKey)
+        {
+            var giver = await _giverRepository.GetGiverAsync(location, rowKey);
+
+            if (giver?.MatchedRecipient is null)
+            {
+                return NoContent();
+            }
+
+            var recipient = await _recipientRepository.GetRecipientAsync(location, giver.MatchedRecipient);
+
+            var originalMatchedRecipient = giver.MatchedRecipient;
+            var originalMatchedGiver = recipient.MatchedGiver;
+
+            giver.HasConfirmedMatch = false;
+            giver.IsSuggestedMatch = false;
+            giver.MatchedRecipient = null;
+
+            recipient.HasConfirmedMatch = false;
+            recipient.IsSuggestedMatch = false;
+            recipient.MatchedGiver = null;
+
+            try
+            {
+                await _giverRepository.InsertOrReplaceAsync(giver);
+                await _recipientRepository.InsertOrReplaceAsync(recipient);
+
+                if (await _connectionRepository.ConnectionExists(giver, recipient))
+                {
+                    await _connectionRepository.DeleteConnectionAsync(location, recipient.RowKey + "_" + giver.RowKey);
+                }
+            }
+            catch (NullReferenceException)
```

Comment:
The operations in the try block could throw other exceptions than NullReferenceException. Should another exception be thrown, the tables might get faulty data as not all operations are completed.  I suggest you change from NullReferenceException to Exception e, and do a _log.Error call.

Reply:
👍🏼